src/EditorFeatures/Core.Wpf/Workspaces/WpfTextBufferVisibilityTracker.cs

```diff
@@ -112,12 +114,17 @@ public void UnregisterForVisibilityChanges(ITextBuffer subjectBuffer, Action cal
 
         private sealed class VisibleTrackerData : IDisposable
         {
-            public readonly HashSet<Action> Callbacks = new();
             public readonly HashSet<ITextView> TextViews = new();
 
             private readonly WpfTextBufferVisibilityTracker _tracker;
             private readonly ITextBuffer _subjectBuffer;
 
+            /// <summary>
+            /// The callbacks that want to be notified when our <see cref="TextViews"/> change visibility.  Stored as an
+            /// <see cref="ImmutableHashSet{T}"/> so we can enumerate it safely without it changing underneath us.
+            /// </summary>
+            public ImmutableHashSet<Action> Callbacks = ImmutableHashSet<Action>.Empty;
```

Comment:
The actual line this impacts is https://github.com/dotnet/roslyn/pull/61671/files#diff-79b25f4e4cd83abb8562b182bca60ef8c82335a76f2735897bcd644f82dd2b73R184, where we enumerate the set.